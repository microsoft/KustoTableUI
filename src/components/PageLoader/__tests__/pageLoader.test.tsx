import renderer, { ReactTestRenderer } from 'react-test-renderer';
import { PageLoader } from '..';

describe('PageLoader Snapshot UI Tests', () => {
  let component: ReactTestRenderer;

  afterEach(() => component.unmount());

  test.each`
    message
    ${''}
    ${undefined}
    ${'Override The Message...'}
  `('Test PageLoader Rendering $scenario', ({ message }) => {
    component = renderer.create(<PageLoader message={message} />);
    const pageLoader = component.toJSON();
    expect(pageLoader).toMatchSnapshot();
  });
});
