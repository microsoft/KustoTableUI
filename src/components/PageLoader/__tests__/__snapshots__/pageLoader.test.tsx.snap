// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLoader Snapshot UI Tests Test PageLoader Rendering $scenario 1`] = `
<div
  className="page-loader"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
  <div />
</div>
`;

exports[`PageLoader Snapshot UI Tests Test PageLoader Rendering $scenario 2`] = `
<div
  className="page-loader"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
  <div>
    Loading...
  </div>
</div>
`;

exports[`PageLoader Snapshot UI Tests Test PageLoader Rendering $scenario 3`] = `
<div
  className="page-loader"
>
  <span
    className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </span>
  <div>
    Override The Message...
  </div>
</div>
`;
